prober/http.go

```diff
@@ -64,6 +66,58 @@ func matchRegularExpressions(reader io.Reader, httpConfig config.HTTPProbe, logg
 	return true
 }
 
+func matchCelExpressions(ctx context.Context, reader io.Reader, httpConfig config.HTTPProbe, logger *slog.Logger) bool {
+	body, err := io.ReadAll(reader)
+	if err != nil {
+		logger.Error("Error reading HTTP body", "err", err)
+		return false
+	}
+
+	bodyJSON := make(map[string]interface{})
```

Comment:
What if the response is a JSON list or a string (or even just `0` or `null`)? It feels like this should be declared as `any` (which matches the `interface{}` where `body` is passed below). Probably worth adding a test for each type as well, but it looks like it should work, provided the CEL expression expects the right type.

```suggestion
	var bodyJSON any
```